bitnami/external-dns/templates/deployment.yaml

```
@@ -652,7 +652,11 @@ spec:
         # TransIP volume(s)
         - name: transip-api-key
           secret:
-            name: {{ template "external-dns.fullname" . }}
+            {{- if .Values.transip.apiKeySecret }}
+            secretName: {{ .Values.transip.apiKeySecret }}
+            {{- else }}
+            secretName: {{ template "external-dns.fullname" . }}
+            {{- end }}
```

Use:
```suggestion
            secretName: {{ template "external-dns.secretName" . }}
```
And modify the `external-dns.secretName` and `external-dns.createSecret` macros accordingly, see:
- https://github.com/bitnami/charts/blob/master/bitnami/external-dns/templates/_helpers.tpl#L116
- https://github.com/bitnami/charts/blob/master/bitnami/external-dns/templates/_helpers.tpl#L169